pennylane/qnode.py

```diff
@@ -221,7 +221,8 @@ def __init__(
         self.gradient_kwargs = None
         self._tape_cached = False
 
-        self._update_gradient_fn()
+        if self.interface != "auto":
+            self._update_gradient_fn()

```

Comment:
Why was this required? It seems like we're not setting the interface to `None` `if self.diff_method is None` by omitting this call.

Reply:
Because we first need to determine the interface before calling `self._update_gradient_fn()` with "auto".

But you are right! I moved the `self.interface != "auto"` check inside the `self._update_gradient_fn` method.